src/coredump/coredump.c

```diff
@@ -1051,16 +1092,35 @@ static int process_socket(int fd) {
 
                         free(iovec.iov_base);
 
-                        found = cmsg_find(&mh, SOL_SOCKET, SCM_RIGHTS, CMSG_LEN(sizeof(int)));
-                        if (!found) {
-                                cmsg_close_all(&mh);
-                                r = log_error_errno(SYNTHETIC_ERRNO(EBADMSG),
-                                                    "Coredump file descriptor missing.");
-                                goto finish;
+                        found = cmsg_find(&mh, SOL_SOCKET, SCM_RIGHTS, CMSG_LEN(sizeof(int) * 2));
+                        if (found) {
+                                int fds[2] = { -EBADF, -EBADF };
+
+                                memcpy(fds, CMSG_TYPED_DATA(found, int), sizeof(int) * 2);
+
+                                assert(mount_tree_fd < 0);
+
+                                /* Maybe we already got coredump FD in previous iteration? */
+                                safe_close(input_fd);
+
+                                input_fd = fds[0];
+                                mount_tree_fd = fds[1];
+
+                                /* We have all FDs we need let's take a shortcut here. */
+                                break;
+                        } else {
+                                /* It is not an array of FDs so maybe it is older version of systemd-coredump sending us just input_fd. */
```

Comment:
If mount_fd is < 0, this is also reached? So it's not exclusive for "older version of systemd-coredump"?

Reply:
Well that is why comment says "maybe it is ..." because that would quite common case for this condition.

Reply:
It's misleading here because it's certain not a very common case, especially considering that AccessContainer= defaults to no.